source/position/position_test.go

```diff
@@ -0,0 +1,96 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package position
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/matryer/is"
+)
+
+func Test_ToSDKPosition(t *testing.T) {
+	is := is.New(t)
+
+	p := Position{
+		Type: TypeSnapshot,
+		Snapshot: SnapshotPositions{
+			"orders": {LastRead: 1, SnapshotEnd: 2},
+		},
+		LastLSN: "4/137515E8",
+	}
+
+	sdkPos := p.ToSDKPosition()
+	is.Equal(
+		string(sdkPos),
+		`{"type":1,"snapshot":{"orders":{"last_read":1,"snapshot_end":2}},"last_lsn":"4/137515E8"}`,
+	)
+
+	var panicErr error
+
+	func() {
+		defer func() {
+			if r := recover(); r != nil {
+				panicErr = errors.New(fmt.Sprint(r))
+			}
+		}()
+
+		var p *Position
+		_ = p.ToSDKPosition()
+	}()
+
+	is.True(panicErr != nil)
+	is.Equal(panicErr.Error(), "runtime error: invalid memory address or nil pointer dereference")
```

Comment:
Hmm, what exactly is being tested with this? It reads like a generic test when accessing a nil pointer, nothing position specific?

Reply:
`ToSDKPosition` does not return an error but panics when marshal fails, so not super easy to test in the regular part, just a test guard.

Reply:
> NaN and +/-Inf values will return an [UnsupportedValueError](https://pkg.go.dev/encoding/json#UnsupportedValueError).

and

> Channel, complex, and function values cannot be encoded in JSON. Attempting to encode such a value causes Marshal to return an [UnsupportedTypeError](https://pkg.go.dev/encoding/json#UnsupportedTypeError).

So I think we're safe, we don't have any of those in the position.

Reply:
@lovromazgon remove the test in this case?

Reply:
If you want, yes, I think we can safely remove it.